### PR TITLE
UX: Re-use left sub-menu for Settings and License page on admin dashboard

### LINKED ISSUE #16399: Update settings layout for the admin dashboard
### Problem to solve

Following up from https://github.com/gitpod-io/gitpod/pull/16307#pullrequestreview-1297969338:

> ❗ issue: One UX change I find a bit jarring is the License and Settings layout, as expected since they were designs with a different layout.

Cc @Devansu-Yadav

### Proposal

> ...
> Reuse the left sidebar which is commonly used for project or org settings, to host two admin settings: General (Usage Statistics and Telemetry settings) and License, so that these pages can continue to use the previous layout.

### PR DESCRIPTION
**Pull request by Devansu-Yadav**
## Description

- Updates the Settings and License page layout to re-use the left sub-menu for better UX.

## Related Issue(s)

Fixes #16399 

## Release Notes

```release-note
Update the Settings and License page layouts for the admin dashboard
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by Devansu-Yadav** [comment]
@gtsiolis Do we need to make the Settings and License pages on the admin dashboard mobile responsive?

**Comment by gtsiolis** [comment]
> Do we need to make the Settings and License pages on the admin dashboard mobile responsive?

@Devansu-Yadav No, it's ok to leave these as they are for now for the following reasons:
1. The changes in this PR don't need to increase the scope further.
2. Our focus will shift away from the admin dashboard.
3. The license page will probably be dropped, see [relevant comment](https://github.com/gitpod-io/gitpod/issues/16399#issuecomment-1431225711).
4. The _General_, previously known as _Settings_ page should be already quite responsive by relying on the existing layout.
6. Responsive design has been in our radar but not a priority so far because of the main targe audience. I'm planning to discuss with @selfcontained next week more about areas we can improve around navigation, etc. in https://github.com/gitpod-io/gitpod/issues/4050. Please forward or open any issues that seem related to that epic. 🍊 

/werft run with-preview-true

**Comment by Devansu-Yadav** [comment]
> 4. The _General_, previously known as _Settings_ page should be already quite responsive by relying on the existing layout.

Yes, it is quite responsive already, except for the Tabs component below the header which is not that responsive due to the no of tabs on the admin dashboard.

> 5. Responsive design has been in our radar but not a priority so far because of the main target audience. I'm planning to discuss with @selfcontained next week more about areas we can improve around navigation, etc. in [Epic: Improve dashboard responsive layout #4050](https://github.com/gitpod-io/gitpod/issues/4050). Please forward or open any issues that seem related to that epic. 🍊

Sure, in fact, I'd _love_ to help out with this issue in case there haven't been any changes addressing the responsive layout for the dashboard 😄

File: components/dashboard/src/admin/admin.routes.ts [hunk 1]
```diff
@@ -28,13 +28,23 @@ export function getAdminTabs(): TabEntry[] {
...
+
+export function getAdminSettingsMenu() {
+    return [
+        {
+            title: "General",
+            link: ["/admin/settings"],
+        },
+        {
+            title: "License",
+            link: ["/admin/license"],
```
Comment (gtsiolis) [inline components/dashboard/src/admin/admin.routes.ts:47]: **issue**: The Admin menu becomes inactive when navigating to the license page. Could we keep it active?

**Review by gtsiolis — CHANGES_REQUESTED** [review]
UX LGTM!

Left one minor comment. If that's not trivial to resolve, let's merge this and open a follow-up issue or PR about it. 🏓

File: components/dashboard/src/admin/admin.routes.ts (hunk 1 above)
Comment (Devansu-Yadav) [reply components/dashboard/src/admin/admin.routes.ts:47]: Fixed it! Was just a one-liner fix that was needed 😆
Actually, I completely missed out that updating the `getAdminTabs` function to remove the License page route might cause such an issue, as this function is being used to set the active state on the admin menu 😅.

**Review by gtsiolis — APPROVED** [review]
UX LGTM, @Devansu-Yadav! 🌮 🌮

File: components/dashboard/src/menu/Menu.tsx [hunk 2]
```diff
@@ -68,7 +68,15 @@ export default function Menu() {
...
-        alternatives: [...getAdminTabs().map((entry) => entry.link)],
+        alternatives: [
+            ...getAdminTabs().reduce(
+                (prevEntry, currEntry) =>
+                    currEntry.alternatives
+                        ? [...prevEntry, ...currEntry.alternatives, currEntry.link]
+                        : [...prevEntry, currEntry.link],
+                [] as string[],
+            ),
+        ],
```
[the remark this answers is not in the document]
Comment (Devansu-Yadav) [reply components/dashboard/src/menu/Menu.tsx:79]: Sure, no problem! 
Basically, this code populates all the top-level admin dashboard routes obtained through the `getAdminTabs` function, which are then passed to the [`isSelected`](https://github.com/gitpod-io/gitpod/pull/16427/files/e8c63abe897bd0b2a42058c30a676f47c8c085fc#diff-f90e2ca98721f64b76c3e081b435071c14de75780931acd6a30d8483933ed1dcR119) function to set the active state for the admin menu on these top-level routes. Hope this makes it a bit clear 😅

[the remark this answers is not in the document]
Comment (gtsiolis) [reply components/dashboard/src/menu/Menu.tsx:79]: Thanks for clarifying, @Devansu-Yadav! 🙇

**Comment by Devansu-Yadav** [comment]
@gtsiolis Based on your [comment](https://github.com/gitpod-io/gitpod/issues/16408#issuecomment-1431156595) in #16408, I was probably looking to pick up one of the following issues - #10499, #3836 or #3930. Should I proceed with these issues, or could you suggest any other issues related to the dashboard component?

**Pull request merged**